restapi/src/main/java/io/stargate/web/docsapi/service/query/condition/ConditionProviderService.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package io.stargate.web.docsapi.service.query.condition;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import io.stargate.web.docsapi.exception.DocumentAPIRequestException;
+import io.stargate.web.docsapi.service.query.condition.provider.ConditionProvider;
+import io.stargate.web.docsapi.service.query.condition.provider.impl.BasicConditionProvider;
+import io.stargate.web.docsapi.service.query.condition.provider.impl.ExistsConditionProvider;
+import io.stargate.web.docsapi.service.query.condition.provider.impl.ListConditionProvider;
+import io.stargate.web.docsapi.service.query.filter.operation.impl.*;
+import java.util.*;
+
+/**
+ * Simple service that wraps all available raw filter values and connects them to a {@link
+ * ConditionProvider}.
+ *
+ * <p>Currently uses a unmodifiable map, thus it's thread safe.
+ */
+public class ConditionProviderService {
```

Comment:
nit: how about `ConditionParser`? "Service" in Stargate is historically something that responds to client requests directly... but it's up to you.

Reply:
I like it, done.